### PR TITLE
Update examples for mapping constructor changes

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
$title.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/21101

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: examples/maps/maps.bal [hunk 1]
```diff
@@ -1,43 +1,73 @@
...
+    string country = "Sri Lanka";
+    string codeLiteral = "code";
+
+    // You can also declare and initialize a `map` with a mapping constructor
+    // expression.
     map<string> addrMap = {
+        // A field can be specified as a key-value pair.
         line1: "No. 20",
         line2: "Palm Grove",
+        // The value in a key-value pair can be any expression that evaluates
```
Comment (praneesha) [inline examples/maps/maps.bal]:
```suggestion
        // The value in a key-value pair can be any expression, which evaluates
```

File: examples/maps/maps.bal [hunk 2]
```diff
@@ -1,43 +1,73 @@
...
     };
     io:println(addrMap);
 
-    // Retrieves a value using member access.
-    string country = <string>addrMap["country"];
-    io:println(country);
+    // Retrieve a value using member access.
+    // Member access returns the value if a field exists in the map with the
+    // specified key, or `()` if a field does not exist with the specified key.
+    // The type of a member access expression for a map is thus the union of
```
Comment (praneesha) [inline examples/maps/maps.bal]:
```suggestion
    // Thus, the type of a member access expression for a map is the union of
```
having "thus" in the middle of the sentence disturbs the flow of it. Better to have it at the beginning.

File: examples/maps/maps.bal [hunk 3]
```diff
@@ -1,43 +1,73 @@
...
+        [codeLiteral]: "00300"
     };
     io:println(addrMap);
 
-    // Retrieves a value using member access.
-    string country = <string>addrMap["country"];
-    io:println(country);
+    // Retrieve a value using member access.
+    // Member access returns the value if a field exists in the map with the
+    // specified key, or `()` if a field does not exist with the specified key.
```
Comment (praneesha) [inline examples/maps/maps.bal]:
```suggestion
    // specified key. Else, it returns `()` if a field does not exist with the specified key.
```
Breaking the text into smaller sentences to increase understandability.

File: examples/maps/maps.bal [hunk 4]
```diff
@@ -1,43 +1,73 @@
...
+    } else {
+        io:println("key 'country' not found");
+    }
 
-    // Retrieves a value using the `.get()` method.
-    // Panics if the map does not have a member with the specified key.
-    string line2 = addrMap.get("line2");
-    io:println(line2);
+    // Retrieve a value using the `.get()` method.
+    // If the map has a field with the specified key, `.get()` returns the value,
```
Comment (praneesha) [inline examples/maps/maps.bal]:
```suggestion
    // If the map has a field with the specified key, `.get()` returns the value.
```

File: examples/maps/maps.bal [hunk 5]
```diff
@@ -1,43 +1,73 @@
...
+    // or panics if the map does not have a field with the specified key.
```
Comment (praneesha) [inline examples/maps/maps.bal]:
```suggestion
    // Else, it panics if the map does not have a field with the specified key.
```

File: examples/maps/maps.bal [hunk 6]
```diff
@@ -1,43 +1,73 @@
...
+    string line2Value = addrMap.get("line2");
+    io:println(line2Value);
 
-    // The `.hasKey()` function checks if a map contains a specified key.
+    // The `.hasKey()` function checks if a map contains the specified key.
     boolean hasPostalCode = addrMap.hasKey("postalCode");
     io:println(hasPostalCode);
 
-    // Adds or updates the value of a key.
+    // Member access lvalue expressions can be used to add fields to a map
```
Comment (praneesha) [inline examples/maps/maps.bal]:
```suggestion
    // Member access value expressions can be used to add fields to a map
```

File: examples/maps/maps.bal [hunk 7]
```diff
@@ -1,43 +1,73 @@
...
+    // or update an already existing field in the map.
```
Comment (praneesha) [inline examples/maps/maps.bal]:
```suggestion
    // or update an already-existing field in the map.
```

File: examples/maps/maps.bal [hunk 8]
```diff
@@ -56,15 +87,13 @@ public function main() {
...
-    map<string> stringMap = {};
-    stringMap["index"] = "100892N";
 
-    // The return type of member access will be `T?` where `T` is the constraint type of the map.
-    // If the key does not exist, nil (`()`) is returned.
-    // The Elvis operator `?:` is a conditional operator that handles `()`.
-    // If the given expression evaluates to nil, the second expression is evaluated and its value is returned.
-    string index2 = stringMap["index"] ?: "";
-    io:println(index2);
+    // A mapping constructor expression can also include a spread field,
```
Comment (praneesha) [inline examples/maps/maps.bal]:
```suggestion
    // A mapping constructor expression can also include a spread field
```
We do not need a comma here.

File: examples/records/records.bal [hunk 9]
```diff
@@ -28,12 +28,21 @@ type Grades record {|
...
+
     // This creates a `Student` record. Since all the fields are required and none of the fields
     // have explicit default values assigned to them, values must be specified for all the fields
     // when creating the record.
     Student john = {
+        // A field can be specified as a key-value pair.
         name: "John Doe",
-        age: 17,
+        // A variable reference can also be used to define a field.
+        // The name of the variable will be used as the key, while
```
Comment (praneesha) [inline examples/records/records.bal]:
```suggestion
        // The name of the variable will be used as the key while
```
We do not need a comma before the "while" conjunction.

File: examples/records/records.bal [hunk 10]
```diff
@@ -89,4 +99,33 @@ public function main() {
     // record, else `()` is returned.
     int? english = grades["english"];
     io:println(english);
+
+    // A mapping constructor expression used when creating a record value
+    // can also include a spread field, referring to another mapping value.
```
Comment (praneesha) [inline examples/records/records.bal]:
```suggestion
    // can also include a spread field referring to another mapping value.
```

File: examples/records/records.bal [hunk 11]
```diff
@@ -89,4 +99,33 @@ public function main() {
...
+        grades: {
+            maths: 70,
+            physics: 80,
+            chemistry: 55
+        },
+        ...address
+    };
+    io:println(anne);
+
+    // Using a mapping constructor expression with `var` (i.e., no contextually expected
```
Comment (praneesha) [inline examples/records/records.bal]:
```suggestion
    // Using a mapping constructor expression with `var` (i.e., no contextually-expected
```

File: examples/records/records.bal [hunk 12]
```diff
@@ -89,4 +99,33 @@ public function main() {
...
+    // type) results in a mapping value where the inferred type is a record type based
```
Comment (praneesha) [inline examples/records/records.bal]:
```suggestion
    // type) results in a mapping value, where the inferred type is a record type based
```
We can have a comma before the "where" conjunction here to increase clarity and understandability since the sentence is long.

File: examples/records/records.bal [hunk 13]
```diff
@@ -89,4 +99,33 @@ public function main() {
...
+    // on the fields specified in the mapping constructor expression.
+    var rec = {name: "Amy", age: 18, ...address};
+    io:println(rec);
+
+    // The record type inferred for `rec` is
+    // `record {| string name; int age; string city; string country; |}`
```
Comment (praneesha) [inline examples/records/records.bal]:
```suggestion
    // `record {| string name; int age; string city; string country; |}`.
```

File: examples/maps/maps.bal (hunk 6 above)
Comment (MaryamZi) [reply examples/maps/maps.bal]: This is a term used in the spec. Added it to distinguish left hand side member access expressions vs right hand side.

File: examples/maps/maps.bal [hunk 14]
```diff
@@ -1,43 +1,73 @@
...
+        // For a computed key, the key expression is evaluated at runtime and
+        // the resulting value is used as the key.
+        [codeLiteral]: "00300"
     };
     io:println(addrMap);
 
-    // Retrieves a value using member access.
-    string country = <string>addrMap["country"];
-    io:println(country);
+    // Retrieve a value using member access.
```
Comment (pubudu91) [inline examples/maps/maps.bal]: This first sentence sounds like a heading. Shall we rephrase this or remove it altogether?

File: examples/maps/maps.bal (hunk 6 above)
Comment (pubudu91) [reply examples/maps/maps.bal]: If we are using the term, we'll need to tell what an lvalue is too right? Should we have a separate BBE for that?

File: examples/maps/maps.bal [hunk 15]
```diff
@@ -1,43 +1,73 @@
...
+    }
 
-    // Retrieves a value using the `.get()` method.
-    // Panics if the map does not have a member with the specified key.
-    string line2 = addrMap.get("line2");
-    io:println(line2);
+    // Retrieve a value using the `.get()` method.
+    // If the map has a field with the specified key, `.get()` returns the value.
+    // Else, it panics if the map does not have a field with the specified key.
+    // Thus, the type for `.get()` on a map is the map's constraint type.
```
Comment (pubudu91) [inline examples/maps/maps.bal]:
```suggestion
    // Thus, the return type of `get()` when invoked on a map is the map's constraint type.
```

File: examples/maps/maps.bal [hunk 16]
```diff
@@ -1,43 +1,73 @@
...
     io:println(hasPostalCode);
 
-    // Adds or updates the value of a key.
+    // Member access lvalue expressions can be used to add fields to a map
+    // or update an already-existing field in the map.
     addrMap["postalCode"] = "00300";
     io:println(addrMap);
 
-    // The `keys()` method returns the keys of the map as an array.
+    // The `.keys()` method returns the keys of the map as an array.
```
Comment (pubudu91) [inline examples/maps/maps.bal]: Lang lib functions are referred to as functions right? Not methods? Also, when referring to functions we haven't used periods in other places IIRC

File: examples/maps/maps.bal [hunk 17]
```diff
@@ -46,8 +76,9 @@ public function main() {
     });
 
     map<int> marks = {sam: 50, jon: 60};
+
     // Calling the `.entries()` method on a map will return the key (`string`)
-    // and the value as a tuple variable.
+    // and the value pairs as tuples.
```
Comment (pubudu91) [inline examples/maps/maps.bal]: "will return the key-value pairs as an array of tuples"

File: examples/records/records.bal [hunk 18]
```diff
@@ -89,4 +99,33 @@ public function main() {
...
+            physics: 80,
+            chemistry: 55
+        },
+        ...address
+    };
+    io:println(anne);
+
+    // Using a mapping constructor expression with `var` (i.e., no contextually-expected
+    // type) results in a mapping value, where the inferred type is a record type based
+    // on the fields specified in the mapping constructor expression.
```
Comment (pubudu91) [inline examples/records/records.bal]:
```suggestion
    // on the types of the fields specified in the mapping constructor expression.
```

File: examples/maps/maps.bal (hunk 6 above)
Comment (MaryamZi) [reply examples/maps/maps.bal]: I'm not quite sure, seems a bit advanced for a BBE. Let me rephrase this anyway.

File: examples/maps/maps.bal (hunk 16 above)
Comment (MaryamZi) [reply examples/maps/maps.bal]: I think I've seen it used both ways. Shouldn't we stick to method when we are invoking it **on** something?

Regarding the periods, again, we seem to have used it both ways. For example, every other comment in this example has a period. I guess we'll have to revisit all of it.

I personally prefer including the period since it indicates we're calling it on something, and it cannot be called on it's own.

Comment (pubudu91) [reply examples/maps/maps.bal]: Yeah, that thought occurred to me as well. But there isn't a particular term used to refer to lang lib functions in the spec, and then again uses methods to refer to object methods and resource methods
